### PR TITLE
[Booking]: made check for dates in method createBooking

### PR DESCRIPTION
**Pull request by DPotapov93**

### CODE REVIEW

File: src/main/java/com/example/accommodationbookingapp/repository/booking/BookingRepository.java [hunk 1]
```diff
@@ -33,4 +33,15 @@ List<Booking> findAllByUserIdAndCheckInDateIsAfter(Long userId,
 
     @Query("SELECT b FROM Booking b JOIN FETCH b.accommodation WHERE b.id = :bookingId")
     Optional<Booking> findBookingWithAccommodationById(@Param("bookingId") Long bookingId);
+
+    @Query("SELECT count(*) > 0 FROM Booking b WHERE b.accommodation.id = :accommodationId AND ("
+            + "            :checkInDate BETWEEN b.checkInDate AND b.checkOutDate"
+            + "            OR :checkOutDate BETWEEN b.checkInDate AND b.checkOutDate"
+            + "            OR b.checkInDate BETWEEN :checkInDate AND :checkOutDate"
+            + "            OR b.checkOutDate BETWEEN :checkInDate AND :checkOutDate"
+            + "    ) ")
```
Comment (vadymhrnk) [inline src/main/java/com/example/accommodationbookingapp/repository/booking/BookingRepository.java:42]:
```suggestion
    @Query("""
        SELECT count(*) > 0 
        FROM Booking b 
        WHERE b.accommodation.id = :accommodationId AND (
                :checkInDate BETWEEN b.checkInDate AND b.checkOutDate
                OR :checkOutDate BETWEEN b.checkInDate AND b.checkOutDate
                OR b.checkInDate BETWEEN :checkInDate AND :checkOutDate
                OR b.checkOutDate BETWEEN :checkInDate AND :checkOutDate
        )
    """)
```

**Review by vadymhrnk — APPROVED** [review]
LGTM. One suggestion for you to consider

**Review by annton-shakur — APPROVED** [review]
GJ

**Review by TarasYurkovskyi — APPROVED** [review]
GJ!

**Pull request merged**